lua/autorun/server/sv_spawn_protection.lua

```diff
@@ -188,81 +176,80 @@ local function setSpawnProtectionForPvpSpawn( ply )
     setLastSpawnTime( ply )
     setSpawnProtection( ply )
     setPlayerTransparent( ply )
-    setPlayerNoCollide( ply )
     createDecayTimer( ply )
 end
 
+-- Instantly removes spawn protection and removes timers and alpha level.
+local function instantRemoveSpawnProtection( ply, message )
+    if not playerHasSpawnProtection( ply ) then return end 
+    removeSpawnProtection( ply, message )
+    setPlayerVisible( ply )
+    removeDecayTimer( ply )
+    removeDelayedRemoveTimer( ply )
+end
+
 -- Called on weapon change to check if the weapon is allowed,
 -- and remove spawn protection if it's not
-local function spawnProtectionWeaponChangeCheck( ply, oldWeapon, newWeapon )
+local function spawnProtectionWeaponChangeCheck( ply, _, newWeapon )
     if not playerIsInPvp( ply ) then return end
     if not playerHasSpawnProtection( ply ) then return end
     if weaponIsAllowed( newWeapon ) then return end
 
     local lastSpawnTime = ply:GetNWInt( "lastSpawnTime", CurTime() - spawnProtectionWeaponGracePeriod )
     if lastSpawnTime >= CurTime() - spawnProtectionWeaponGracePeriod then return end
 
-    local printMessage = "You've equipped a weapon and lost spawn protection"
-    removeSpawnProtection( ply, printMessage )
-    setPlayerVisible( ply )
-    setPlayerCollide( ply )
-    removeDecayTimer( ply )
-    removeDelayedRemoveTimer( ply )
+    instantRemoveSpawnProtection( ply, "You've equipped a weapon and lost spawn protection." )
+end
+
+-- Remove spawn protection on using objects
+local function spawnProtectionUseCheck( ply )
+    if not playerIsInPvp( ply ) then return end
+    if not playerHasSpawnProtection( ply ) then return end
+    instantRemoveSpawnProtection( ply, "You've used an entity and lost spawn protection." )
```

Comment:
```suggestion
-- Remove spawn protection on using objects
local function spawnProtectionUseCheck( ply, ent )
    if not playerIsInPvp( ply ) then return end
    if not playerHasSpawnProtection( ply ) then return end
    if not IsValid( ent ) then return end
    if not ent.Use then return end

    instantRemoveSpawnProtection( ply, "You've used an entity and lost spawn protection." )
```

Maybe we should make sure the ent can be used and has a Use function?

Reply:
I just used this as you can still pick stuff up with USE or use doors and such. Felt like those should disable it too.

Reply:
Sure, we'll see how it works